src/ui/ui.ts

```diff
@@ -10,21 +10,48 @@ function onOpen(): void {
 }
 
 function menuStep(): void {
-  step(getLastActiveRowNumber("A", runSheet) - 2);
+  step(getLastActiveRowNumber(`${pcColumn}`, runSheet) - 2);
 }
 
 function menuRun(): void {
-  runUntilPc();
+  let lastStepNumber: number = runUntilPc();
+  if (isProofMode()) {
+    step(lastStepNumber - 1);
+    let equalToLastStepFormulas: string[] = [];
+    for (let i = 0; i <= columnToIndex(runOp1Column); i++) {
+      equalToLastStepFormulas.push(`=${indexToColumn(i)}${lastStepNumber + 1}`);
+    }
+    for (
+      let stepNum = lastStepNumber + 2;
+      stepNum <= nextPowerOfTwo(lastStepNumber) + 1;
+      stepNum++
+    ) {
+      runSheet
+        .getRange(`${pcColumn}${stepNum}:${runOp1Column}${stepNum}`)
+        .setFormulas([equalToLastStepFormulas]);
+    }
+  }
```

Comment:
not sure if it's regular to do it like that. The program needs to end with an infinite loop for proof mode to work, and it just keeps running the program until a given number of steps is done.
The `cairo-compile` tool actually adds this bytecode when used with `--proof-mode`. Alternatively, you could put a jmp rel 0 in final_pc so that it keeps running, and make sure that it's part of the public memory

Reply:
Currently if you run a program that was compiled in proof-mode it will indeed do an infinite loop which repeats the last step. But I figured it was quicker to simply set the last lines as equal to the last execution line. But I can simply do a run until PC and then call step() as much as needed it will simply be slower.

Reply:
at least one need to make sure to have a proper program that does the loop